tests/test_sklearn_decomposition.py

```diff
@@ -0,0 +1,146 @@
+    def test_pca_converter_two(self):
+        self._fit_model_pca(PCA(n_components=2))
+
+    # PCA n_componenets mle and whiten true
+    @unittest.skipIf(
+        LooseVersion(sklearn.__version__) < LooseVersion("0.23.2"),
+        reason="With Sklearn version < 0.23.2 returns ValueError: math domain error (https://github.com/scikit-learn/scikit-learn/issues/4441)",
+    )
+    def test_pca_converter_mle_whiten(self):
+        self._fit_model_pca(PCA(n_components="mle", whiten=True))
+
+    # PCA n_componenets mle and solver full
+    @unittest.skipIf(
+        LooseVersion(sklearn.__version__) < LooseVersion("0.23.2"),
+        reason="With Sklearn version < 0.23.2 returns ValueError: math domain error (https://github.com/scikit-learn/scikit-learn/issues/4441)",
+    )
+    def test_pca_converter_mle_full(self):
+        self._fit_model_pca(PCA(n_components="mle", svd_solver="full"))
+
+    # PCA n_componenets none and solver arpack
+    def test_pca_converter_none_arpack(self):
+        self._fit_model_pca(PCA(n_components=None, svd_solver="arpack"))
+
+    # PCA n_componenets none and solver randomized
+    def test_pca_converter_none_randomized(self):
+        self._fit_model_pca(PCA(n_components=None, svd_solver="randomized"))
+
+    # KernelPCA linear kernel
+    def test_kernel_pca_converter_linear(self):
+        self._fit_model_pca(KernelPCA(n_components=5, kernel="linear"))
+
+    # KernelPCA linear kernel with inverse transform
+    def test_kernel_pca_converter_linear_fit_inverse_transform(self):
+        self._fit_model_pca(KernelPCA(n_components=5, kernel="linear", fit_inverse_transform=True))
+
+    # KernelPCA poly kernel
+    def test_kernel_pca_converter_poly(self):
+        self._fit_model_pca(KernelPCA(n_components=5, kernel="poly", degree=2))
+
+    # KernelPCA poly kernel coef0
+    def test_kernel_pca_converter_poly_coef0(self):
+        self._fit_model_pca(KernelPCA(n_components=10, kernel="poly", degree=3, coef0=10))
+
+    # KernelPCA poly kernel with inverse transform
+    def test_kernel_pca_converter_poly_fit_inverse_transform(self):
+        self._fit_model_pca(KernelPCA(n_components=5, kernel="poly", degree=3, fit_inverse_transform=True))
+
+    # KernelPCA poly kernel
+    def test_kernel_pca_converter_rbf(self):
+        self._fit_model_pca(KernelPCA(n_components=5, kernel="rbf"))
+
+    # KernelPCA sigmoid kernel
+    def test_kernel_pca_converter_sigmoid(self):
+        self._fit_model_pca(KernelPCA(n_components=5, kernel="sigmoid"))
+
+    # KernelPCA cosine kernel
+    def test_kernel_pca_converter_cosine(self):
+        self._fit_model_pca(KernelPCA(n_components=5, kernel="cosine"))
+
+    # KernelPCA precomputed kernel
+    def test_kernel_pca_converter_precomputed(self):
+        self._fit_model_pca(KernelPCA(n_components=5, kernel="precomputed"), precompute=True)
+
+    # TODO: Fails on macos-latest Python 3.8 due to a sklearn bug.
+    # FastICA converter with n_components none
```

Comment:
Can you please open an issue for this so we can fix it up when the bug is fixed?

Reply:
done